### DISCUSSION TITLE
Manuscripts: Can I have notebooks that are rendered as classical Rmarkdown scripts (!=not Jupyter Notebooks) ?

### DISCUSSION THREAD

**Discussion by LGraz** [quarto-dev/quarto-cli#11076] [Q&A]
### Description

**Background:** I want to render the example below in a manuscript and link to it in a nice way. Notebooks seem to be the right option. However, the output below uses some Rmd tricks and things do not work with Jupyter notebooks. 

Is there any possibility that allows me to use the non-jupyter rendered html files?

**Setup:** The file below lives in `notebooks/plotCars.qmd`  and the `_quarto.yml` contains 
```yml
manuscript:
  notebooks:
    - notebook: notebooks/cars-plot.qmd
    ...
```

````qmd
---
title: Cars Plot
author: Lukas Graz
engine: knitr
date: 2024-10-16
---

```{r cars-plot, fig.cap="A scatter plot of the cars data set."}
plot(cars)
library(qreport)
library(ggplot2)
```

```{r, results='asis'}
qreport::maketabs(
  list(
    ggplot(cars, aes(x = speed, y = dist)) + geom_point(),
    ggplot(cars, aes(x = speed, y = dist)) + geom_jitter()
))
```
````

Thanks <3

**Comment by mcanouil** [comment]
You mean have the Quarto document linked? (note that it is not `rmarkdown`)

`code-tools`? https://quarto.org/docs/output-formats/html-code.html#code-tools

`other-links` or `code-links`? https://quarto.org/docs/output-formats/html-basics.html#code-links-and-other-links

---

### Side note


Quarto recommends the use of YAML style for code cells options because Quarto itself does not use R inline syntax. That's supported by Quarto as a side effect with `knitr` as the engine.
So your document would be:

````qmd
---
title: Cars Plot
author: Lukas Graz
engine: knitr
date: 2024-10-16
---

```{r}
#| label: cars-plot
#| fig-cap: "A scatter plot of the cars data set."
plot(cars)
library(qreport)
library(ggplot2)
```

```{r}
#| results: asis
qreport::maketabs(
  list(
    ggplot(cars, aes(x = speed, y = dist)) + geom_point(),
    ggplot(cars, aes(x = speed, y = dist)) + geom_jitter()
))
```
````

**Comment by LGraz** [comment]
Thanks a lot, I was not aware that I can also use all the features from [HTML](https://quarto.org/docs/output-formats/html-code.html). Great!

Yes, I mean a quarto-document linked 

Thanks to you I know how to link a html file, I suppose I just write a pre-render script that renders the file to quarto.

![image](https://github.com/user-attachments/assets/8999e2a6-0325-4db2-a8cf-6b3becbaf917)

Its good but not perfect. I prefer the other-links to appear below the table-of-contents. :)

Ideally, I would just be able to specify in the notebooks setting that I want it not as a Jupyter but a classical quarto document (or "rmarkdown" as I falsley named it)

**Reply by mcanouil** [reply]
I don't understand.
Your input file is a Quarto document or a Jupyter notebook?

`quarto convert` is the command to switch between the two.
Note that the two are not really formats since they should be equivalent or even identical.


What's your workflow?

**Reply by mcanouil** [reply]
> Thanks a lot, I was not aware that I can also use all the features from HTML. Great!

Manuscript is a project type. It does not change the formats.
If your Manuscript uses `pdf` format, then you have access to the related options, etc. for the formats compatible with Manuscript project.

**Reply by LGraz** [reply]
the confusion was that
![image](https://github.com/user-attachments/assets/0d30687c-42f2-43a6-a785-a3e6cc8f224a)
is generated from a *.qmd file.  But the "In [2]: " reminded me of Jupyter. Thanks for the clarification!

In the meantime I found a workaround, so I close it (even though `qreport::maketabs` still does not work though since it is confused with figure-paths).

Thanks again :)

**Reply by mcanouil** [reply]
You'll have to ask the author of this package then.

When `engine: jupyter` (explicitly declared or auto-detected because of Python or multiple languages), it does not change much if the input was Quarto document or a Jupyter notebook as "in the end" it's a Jupyter Notebook.
